### PR TITLE
update recipe to use 2020.1 buildnum 217

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Checklist
* [ X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ X] Bumped the build number (if the version is unchanged)
* [ X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-{% set version = "2020.0" %}
-{% set buildnum = "166" %}
+{% set version = "2020.1" %}
+{% set buildnum = "217" %}
```
Comment (isuruf) [inline recipe/meta.yaml:2]: Looks like this needs to be `217` for linux and `216` for other platforms.

Comment (oleksandr-pavlyk) [reply recipe/meta.yaml:2]: Yes, you are write:

```
(base) [14:44:54 vmlin work]$ conda search -c intel mkl[subdir='linux-64']=2020.1
Loading channels: done
# Name                       Version           Build  Channel
mkl                           2020.1       intel_217  intel
(base) [14:45:16 vmlin work]$ conda search -c intel mkl[subdir='osx-64']=2020.1
Loading channels: done
# Name                       Version           Build  Channel
mkl                           2020.1       intel_216  intel
(base) [14:45:37 vmlin work]$ conda search -c intel mkl[subdir='win-64']=2020.1
Loading channels: done
# Name                       Version           Build  Channel
mkl                           2020.1       intel_216  intel
```

Comment (oleksandr-pavlyk) [reply recipe/meta.yaml]: I tried using 

```
{% set buildnum = "217" if target_platform.startswith('linux') else "216" %}
```

but that failed with 

```
  File "/home/conda/recipe_root/meta.yaml", line 2, in top-level template code
    {% set buildnum = "217" if target_platform.startswith('linux') else "216" %}
TypeError: 'NoneType' object is not callable
```

Comment (isuruf) [reply recipe/meta.yaml]: I think another rerender is needed.

Comment (oleksandr-pavlyk) [reply recipe/meta.yaml]: An attempt to rerender brings out the same error:

```
  File "/tmp/miniconda3/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/workdir/git/intel_repack-feedstock/recipe/meta.yaml", line 2, in top-level template code
    {% set buildnum = "217" if target_platform.startswith('linux') else "216" %}
TypeError: 'NoneType' object is not callable
```

Comment (beckermr) [reply recipe/meta.yaml]: selectors would do the trick here maybe?

**Comment by beckermr** [comment]
Seriously?

**Comment by beckermr** [comment]
maybe a straight if statement in jinja2 - we do this all the time right?

**Comment by oleksandr-pavlyk** [comment]
I tried running `recipe-lint` as suggested and got:

```
No azure token. Create a token and
put it in ~/.conda-smithy/azure.token
Traceback (most recent call last):
  File "/localdisk/work/opavlyk/miniconda3_latest/envs/build_clpy_cf/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/localdisk/work/opavlyk/miniconda3_latest/envs/build_clpy_cf/lib/python3.8/site-packages/conda_smithy/cli.py", line 559, in main
    args.subcommand_func(args)
  File "/localdisk/work/opavlyk/miniconda3_latest/envs/build_clpy_cf/lib/python3.8/site-packages/conda_smithy/cli.py", line 394, in __call__
    lints, hints = lint_recipe.main(
  File "/tmp/miniconda3/envs/base/lib/python3.8/site-packages/conda_smithy/lint_recipe.py", line 762, in main
    with io.open(recipe_meta, "rt") as fh:
FileNotFoundError: [Errno 2] No such file or directory: 
'/work/git/intel_repack-feedstock/meta.yaml'
```

It should be looking in `recipe/meta.yaml`, not `meta.yaml`.

**Comment by beckermr** [comment]
Run conda-forge recipe-lint recipe

**Comment by oleksandr-pavlyk** [comment]
Now:

```
    raise ComposerError(
ruamel.yaml.composer.ComposerError: expected a single document in the stream
  in "<unicode string>", line 2, column 3:
      [linux]
      ^ (line: 2)
but found another document
  in "<unicode string>", line 3, column 3:
      [not linux]
      ^ (line: 3)
```

**Comment by oleksandr-pavlyk** [comment]
I was able to fix this with use of `#`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {% set version = "2020.1" %}
-{% set buildnum = "217" %}  [linux]
-{% set buildnum = "216" %}  [not linux]
+{% set buildnum = "217" %}  # [linux]
+{% set buildnum = "216" %}  # [not linux]
```
Comment (beckermr) [inline recipe/meta.yaml:3]: Ahhhh I cannot read. Selectors have to be `# [stuff]` otherwise they are not proper yaml comments.

**Comment by oleksandr-pavlyk** [comment]
`osx osx_` failed with some connectivity issues.

**Comment by beckermr** [comment]
@conda-forge-admin please restart ci

**Comment by oleksandr-pavlyk** [comment]
I assume an exception to linter's suggestions is warranted here, and hence this can be merged, now that builds are green. 

Thank you

**Comment by isuruf** [comment]
@oleksandr-pavlyk, do you know if this version support LAPACK 3.9.0 API? We'd like to update out stack to LAPACK 3.9.0, but MKL has been the hold out.

**Comment by oleksandr-pavlyk** [comment]
@isuruf According to https://software.intel.com/en-us/forums/intel-math-kernel-library/topic/840653 MKL supports Lapack 3.8.0 API.

I will ask around for the upgrade timeline.

**Comment by isuruf** [comment]
Thanks @oleksandr-pavlyk

**Comment by h-vetinari** [comment]
> @oleksandr-pavlyk: I will ask around for the upgrade timeline.

Thanks! I opened an issue to track this: #6

**Pull request merged**